tasks/main.yml

```diff
@@ -41,41 +41,34 @@
       - name: "{{ 'cpu_affinity'
                   if kernel_settings_systemd_cpu_affinity
                   or kernel_settings_systemd_cpu_affinity_state
-                  else omit }}"
-        value: "{{ kernel_settings_systemd_cpu_affinity
-                   if kernel_settings_systemd_cpu_affinity
-                   else omit }}"
-        state: "{{ kernel_settings_systemd_cpu_affinity_state
-                   if kernel_settings_systemd_cpu_affinity_state
-                   else omit }}"
+                  else none }}"
+        value: "{{ kernel_settings_systemd_cpu_affinity |
+                   d(none, true) }}"
```

Comment:
May I ask why the default value is `none, true` for all?

Reply:
`none` in Ansible/jinja2 is converted to the python `None` value - the module treats this value as empty, and removes it.  I could have used any "empty" value, such as `""` or `[]` or `{}` - but `none` seems clearer.

Reply:
also - `kernel_settings_systemd_cpu_affinity | d(none, true)` - https://jinja.palletsprojects.com/en/2.11.x/templates/#default

the `true` means that the `default` filter will evaluate the given variable in the boolean context if defined - it is a shorthand for `if kernel_settings_systemd_cpu_affinity` - so I can do the `is defined` and "evaluates to true in a boolean context" test in one filter